exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/dml/FetchBatchedResultsTests.kt

```diff
@@ -100,4 +101,26 @@ class FetchBatchedResultsTests : DatabaseTestsBase() {
     fun `when batch size is 0 or less, should throw an exception`() {
         DMLTestsData.Cities.selectAll().fetchBatchedResults(batchSize = -1)
     }
+
+    @Test
+    fun testFetchBatchedResultsWithAutoIncrementEntityId() {
+        val tester1 = object : IntIdTable("table_1") {
+            val data = varchar("data", 100)
+        }
+
+        val tester2 = object : IntIdTable("table_2") {
+            val moreData = varchar("more_data", 100)
+            val prevData = reference("prev_data", tester1, onUpdate = ReferenceOption.CASCADE)
+        }
+
+        withDb {
+            try {
+                SchemaUtils.create(tester1, tester2)
```

Comment:
It might be better to use `withTables(tester1, tester2)` instead of `withDb` in this case.